packages/@headlessui-react/src/components/popover/popover.tsx

```diff
@@ -350,7 +185,18 @@ function PopoverFn<TTag extends ElementType = typeof DEFAULT_POPOVER_TAG>(
   let root = useRootContainers({
     mainTreeNode,
     portals,
-    defaultContainers: [button, panel],
+    defaultContainers: [
+      {
+        get current() {
+          return machine.state.button
+        },
+      },
+      {
+        get current() {
+          return machine.state.panel
+        },
+      },
+    ],
```

Comment:
This one is a bit odd, but using `button` and `panel` directly gives us the wrong state. This makes it lazy.

That said, the values will be used in callbacks and not during render, so I want to clean all that up in a follow up PR.